packages/core/src/Stores/client-store.js

```diff
@@ -229,25 +229,13 @@ export default class ClientStore extends BaseStore {
 
     @computed
     get legal_allowed_currencies() {
-        if (!this.landing_companies) return [];
+        if (!this.landing_companies && !this.current_landing_company) return [];
         if (this.root_store.ui && this.root_store.ui.real_account_signup_target) {
-            const target = this.root_store.ui.real_account_signup_target === 'maltainvest' ? 'financial' : 'gaming';
-            if (this.landing_companies[`${target}_company`] && this.current_landing_company && this.accounts) {
-                if (this.accounts[this.loginid] && !this.accounts[this.loginid].currency) {
-                    return this.landing_companies[`${target}_company`].legal_allowed_currencies.filter(currency =>
-                        this.current_landing_company.legal_allowed_currencies.includes(currency)
-                    );
-                }
-            }
-            if (this.landing_companies[`${target}_company`]) {
-                return this.landing_companies[`${target}_company`].legal_allowed_currencies;
+            if (this.root_store.ui.real_account_signup_target.match(/set_currency|manage/)) {
```

Comment:
```suggestion
            if (['set_currency',  'manage'].includes(this.root_store.ui.real_account_signup_target)) {
```

This way we can have a more strict condition check since the regex could be more restrictive for "possible" future changes.